test/tstQueryTreeRay.cpp

```diff
@@ -130,5 +133,80 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(test_ray_box_intersection, DeviceType,
       exec_space, tree, predicates,
       make_reference_solution<int>({0, 1, 2, 3, 4, 5, 6, 7, 8, 9}, {0, 10}));
 }
+BOOST_AUTO_TEST_SUITE_END()
+
+template <typename DeviceType>
+struct InsertIntersections
+{
+  mutable int _i;
+  Kokkos::View<int *, DeviceType> _ordered_intersections;
+
+  template <typename Predicate>
+  KOKKOS_FUNCTION void operator()(Predicate const &,
+                                  int const primitive_index) const
+  {
+    _ordered_intersections(Kokkos::atomic_fetch_inc(&_i)) = primitive_index;
+  }
+};
+
+template <typename DeviceType>
+struct BoxesIntersectedByRayOrdered
+{
+  Kokkos::View<ArborX::Experimental::Ray *, DeviceType> rays;
+};
+
+template <typename DeviceType>
+struct ArborX::AccessTraits<BoxesIntersectedByRayOrdered<DeviceType>,
+                            ArborX::PredicatesTag>
+{
+  using memory_space = typename DeviceType::memory_space;
+  static KOKKOS_FUNCTION int
+  size(BoxesIntersectedByRayOrdered<DeviceType> const &nearest_boxes)
+  {
+    return nearest_boxes.rays.size();
+  }
+  static KOKKOS_FUNCTION auto
+  get(BoxesIntersectedByRayOrdered<DeviceType> const &nearest_boxes, int i)
+  {
+    return ordered_intersects(nearest_boxes.rays(i));
+  }
+};
+
+BOOST_AUTO_TEST_SUITE(RayTraversals)
+
+BOOST_AUTO_TEST_CASE_TEMPLATE(test_ray_box_intersection_new, DeviceType,
+                              ARBORX_TEST_DEVICE_TYPES)
+{
+  using memory_space = typename DeviceType::memory_space;
+  typename DeviceType::execution_space exec_space;
+
+  std::vector<ArborX::Box> boxes;
+  int const n = 10;
+  for (unsigned int i = 0; i < n; ++i)
+    boxes.emplace_back(ArborX::Point(i, i, i),
+                       ArborX::Point(i + 1, i + 1, i + 1));
+  auto device_boxes = ArborXTest::toView<DeviceType>(boxes, "boxes");
+
+  ArborX::BVH<memory_space> const tree(exec_space, device_boxes);
+
+  ArborX::Experimental::Ray ray{
+      ArborX::Point{0, 0, 0},
+      ArborX::Experimental::Vector{1. / n, 1. / n, 1. / n}};
```

Comment:
Can we please expand this test by shooting a  second ray from `(n, n, n)` with direction `(-1/n, -1/n, -1/n)`?

Reply:
a4e33e37045fdbdfd53695a7715f755e8c794b64